python/lsst/meas/algorithms/detection.py

```diff
@@ -698,8 +699,13 @@ def reEstimateBackground(self, maskedImage, backgrounds):
             self.log.warn("Fiddling the background by %g", self.config.adjustBackground)
             bg += self.config.adjustBackground
         self.log.info("Resubtracting the background after object detection")
-        maskedImage -= bg.getImageF()
-        backgrounds.append(bg)
+        maskedImage -= bg.getImageF(self.background.config.algorithm,
+                                    self.background.config.undersampleStyle)
+
+        actrl = bg.getBackgroundControl().getApproximateControl()
+        backgrounds.append((bg, getattr(afwMath.Interpolate, self.background.config.algorithm),
+                            bg.getAsUsedUndersampleStyle(), actrl.getStyle(), actrl.getOrderX(),
+                            actrl.getOrderY(), actrl.getWeighting()))
```

Comment:
I assume this is the new API, but boy that is a verbose yet opaque call.

Reply:
💯 

I found this totally impenetrable, and only worked it out by a combination of cargo-culting and trial-and-error. I think this is the best support for @kfindeisen's suggestion on [DM-23113](https://jira.lsstcorp.org/browse/DM-23113) that we should just back out these changes altogether. (Which we might still do, but I wouldn't be inclined to do so on a Friday afternoon!)

Reply:
Yeah, `BackgroundList` is exactly what first made me think that maybe I shouldn't be touching this code. 😱